src/js/modules/core/services/sweeper/sweeper-service.factory.js

```diff
@@ -0,0 +1,207 @@
+            })
+        }
+
+        function signAndSweep(resultUTXOs, options) {
+            // Create raw transaction
+            var inputs = [];
+            var totalValue = 0;
+            var addresses = Object.keys(resultUTXOs);
+            var rawTransaction = new bitcoinJS.TransactionBuilder();
+
+            // If Bitcoin Cash sweep
+            if (options.network.toLowerCase() == 'bcc') {
+                rawTransaction.enableBitcoinCash(true);
+            }
+
+            // For each address (key) of res object, the value is an array of UTXO objects
+            for (var i = 0; i < addresses.length; i++) {
+                var address = addresses[i];
+                for (var utxoIndex in resultUTXOs[address]) {
+
+                    // If index is integer (UTXO)
+                    if (utxoIndex == parseInt(utxoIndex, 10)) {
+                        var utxo = resultUTXOs[address][utxoIndex];
+
+                        // Add to totalValue
+                        totalValue += parseInt(utxo['value']);
+
+                        rawTransaction.addInput(utxo['hash'], utxo['index']);
+                        inputs.push({
+                            txid: utxo['hash'],
+                            vout: utxo['index'],
+                            scriptPubKey: utxo['script_hex'],
+                            value: utxo['value'],
+                            address: address
+                        });
+                    }
+                }
+            }
+
+            return bitcoinDataClient.estimateFee().then(function (feePerKB) {
+                var approxTxSize = estimateApproximateFeeP2PKH(inputs.length, 1);
+                var approxFee = Math.floor(approxTxSize * feePerKB / 1000);
+
+                var outputValue = totalValue - approxFee;
+
+                if (outputValue > 547) {
+                    rawTransaction.addOutput(options.recipient, outputValue);
+
+                    angular.forEach(inputs, function (input, index) {
+                        var currentPrivKey = resultUTXOs[input.address].priv_key;
+
+                        // Set hashType depending on network
+                        var hashType = bitcoinJS.Transaction.SIGHASH_ALL | (options.network.toLowerCase() == 'bcc' ? bitcoinJS.Transaction.SIGHASH_BITCOINCASHBIP143 : 0);
+
+                        rawTransaction.sign(index, currentPrivKey, null, hashType, input.value);
+                    });
+
+                    return {
+                        rawTx: rawTransaction.build().toHex(),
+                        feePaid: approxFee,
+                        inputCount: inputs.length,
+                        totalValue: totalValue
+                    }
+                } else {
+                    return false;
+                }
+            });
+        }
+
+        function hdDiscover(root, batchSize, accountBatchSize, testnet, accountIdx, UTXOs) {
+            testnet = testnet || 0;
```

Comment:
The testnet parameter is passed as a boolean in some places, but here it defaults to 0, and it's only used as a string in the paths..